applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.cpp

```
@@ -0,0 +1,547 @@
+
+double& PlasticityIsotropicKinematicJ2::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+         const Variable<double>& rThisVariable,
+                         double& rValue)
+{
+    if(rThisVariable == EQUIVALENT_PLASTIC_STRAIN){
+        rValue = mEquivalentPlasticStrain;
+    }
+    return(rValue);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
+{
+    FinalizeMaterialResponseCauchy(rValues);  // not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    FinalizeMaterialResponseCauchy(rValues);  // not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues)
+{
+    FinalizeMaterialResponseCauchy(rValues);  // not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    BoundedArrayType plastic_strain;
+    double           equivalent_plastic_strain;
+    
+    switch ( VoigtSize )
+    {
+    case 3:
+        //this->CalculateResponse3(rValues, plastic_strain, equivalent_plastic_strain);
```

still you should throw an error if not implemented